src/index.ts

```
@@ -0,0 +1,45 @@
+import { v4 as uuidv4 } from 'uuid';
+
+type AlgorithmFunction = () => string;
+
+const algorithms: Record<string, AlgorithmFunction> = {
+  uuid4: uuidv4,
+};
+
+function replacingAlgorithm(
+  memoizationTable: Record<string, string>,
+  _: unknown,
+  uriBase: string,
+  uuidAlgorithm: string,
+  originalUuid: string,
+) {
+  if (memoizationTable[`${uriBase}${originalUuid}`]) {
+    return memoizationTable[`${uriBase}${originalUuid}`];
+  } else {
+    const algorithm = algorithms[uuidAlgorithm] || uuidv4;
```

Should it default to `uuidv4`? It seems like this could hide problems from missing algorithms.

---

This relates to the other question, in this case I also opted for the silent failing, and do the best with what we have, if the user specifies an algorithm that we don't have we just use uuidv4, another option would be to hard fail and throw an error, we can discuss it on the standup